runtime/scripts/math.js

```diff
@@ -2051,6 +2051,124 @@ var math = Numbas.math = /** @lends Numbas.math */ {
             }
         }
         return factors;
+    },
+           /** Combines two matrices horizontally
+     * 
+     * @param {matrix} m1
+     * @param {matrix} m2
+     * @returns {matrix}
+     */
+            combine_horizontally: function(m1,m2){
+                m1.columns = m1[0].length;
+                m1.rows = m1.length;
+        
+                m2.columns = m2[0].length;
+                m2.rows = m2.length;
+        
+                for(var i = 0; i < m2.columns+1; i++) {
+        
+                    if (m2.rows > m1.rows){
+                        for(var j=m1.rows;j<m2.rows;j++){
+                        m1[m1.rows]=[]
+                            for (var x = 0; x < m1.columns; x++){
+                                m1[m1.rows].push('0');
+                            }
+                        m1.rows++
+                    }
+                    }
+                    else if (m2.rows < m1.rows){
+                        for(var j=m2.rows;j<m1.rows;j++){
+                            m2[m2.rows]=[]
+                        for (var y = 0; y < m2.columns; y++){
+                            m2[m2.rows].push('0');
+                        }
+                        m2.rows++
+                    }
+                    }
+                }
+                var m3=m1.concat(m2)
+                m3.rows=Math.max(m1.rows, m2.rows)
+                m3.columns=m1.columns+m2.columns
+                return m3
```

Comment:
While `m3` is a new object, each of the rows will be returned unchanged. Since arrays in JS are mutable, that could lead to weird behaviour if something else then goes on to change `m1` or `m2`.
Either `.concat` or `.slice` would produce new array objects for the rows.